src/snowflake/cli/api/project/definition_manager.py

```diff
@@ -100,5 +101,5 @@ def _user_definition_file_if_available(project_path: Path) -> Optional[Path]:
         )
 
     @functools.cached_property
-    def project_definition(self) -> dict:
+    def project_definition(self) -> ProjectDefinition:
```

Comment:
Love the type safety Pydantic gives us!

Reply:
Me too :)
As i cannot answer the comment above - yes, Pydantic can be used to generate JSON schemas, but i think i'll summon @sfc-gh-turbaszek to answer the long-term vision plans :D

Reply:
And we could use `description` and `examples` fields of FieldInfo, to have the schema well documented.